main.ts

```diff
@@ -122,6 +122,23 @@ class Translator {
     return true;
   }
 
+  visitNamedParameter(paramDecl: ts.ParameterDeclaration) {
+    this.visitEachIfPresent(paramDecl.decorators);
+    if (paramDecl.type) {
+      // TODO(martinprobst): These are currently silently ignored.
+      // this.reportError(paramDecl.type, 'types on named parameters are unsupported');
+    }
+    this.visit(paramDecl.name);
+    if (paramDecl.initializer) {
```

Comment:
actually that's going sort of towards a linter, but should this also fail if there is no initializer for a named parameter list? That's AFAICT always wrong (fails if you don't pass any named parameters).